### DISCUSSION TITLE
⚡ Copilot Metrics API Organization Team Metrics 👥

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#135502] [Copilot News and Announcements]
We’re excited to share that usage metrics for GitHub Organization Teams are now available on the public beta of the [GitHub Copilot Metrics API](https://docs.github.com/en/rest/copilot/copilot-usage?apiVersion=2022-11-28)! 🎉 

### What metrics are available for GitHub Organization Teams?

*   ⭐ Organization Team aggregates are available for teams with five or more Copilot license holders.
*   ⭐ Teams must belong to the GitHub Organization which provisioned team members’ licenses.
*   ⭐ The beta of the GitHub Copilot Metrics API is focused on serving metrics for Copilot Chat and code completions that take place in the IDE.
*   ⭐ **Code completion metrics include**: Lines of Code Suggested, Lines of Code Accepted, Number of Suggestions, Number of Acceptances, and Active Users, with slices on language and IDE.
*   ⭐ **Copilot Chat metrics include**: Number of Chats, Chat Suggestions Accepted, and Active Users. The endpoint does not currently feature slices on language or IDE for Chat metrics.

### Documentation and Resources

See the following resources for help getting started:  
- **API Documentation**: Explore the detailed API documentation, including metrics definitions [here](https://docs.github.com/en/rest/copilot/copilot-usage?apiVersion=2022-11-28).  
- **Learning Pathway**: You can find an extended article on measuring the impact of GitHub Copilot [here](https://resources.github.com/learn/pathways/copilot/essentials/measuring-the-impact-of-github-copilot/).

### Participate in the Public Beta!

Your feedback during this beta phase is invaluable to us. We encourage you to share your experiences, which will be instrumental in refining and enhancing the API as we look forward to the GA release. Please share your feedback in the comments below 👇

**Comment by kkc-tonywu** [comment]
Hi @Akash1134,

When I tried using the new Copilot Metrics API for Organization Team Metrics, I kept receiving empty data.

Reproduction Steps:

Run the following `curl` command:
```
$ curl --location 'https://api.github.com/orgs/<org>/team/<team>/copilot/usage' \ 
       --header 'Accept: application/vnd.github+json' \ 
       --header 'Authorization: Bearer ghp_xxxx' \ 
       --header 'X-GitHub-Api-Version: 2022-11-28' 
```

The response was:
```
[]
```
I’m quite certain that `<org>`, `<team>`, and the access token are all correct.

Why do I keep getting an empty array when I try? What might be going wrong?

**Reply by Akash1134** [reply]
Hi @kkc-tonywu ,

Thanks for reporting. I have shared it with my team internally.

**Reply by milos-licina-3shape** [reply]
This happens to me also. [The documentation](https://docs.github.com/en/rest/copilot/copilot-usage?apiVersion=2022-11-28#get-a-summary-of-copilot-usage-for-a-team) states that **"Enterprise administration" business permissions (read)** are required but there seems to be no way of setting that permission.

**Reply by fortmate** [reply]
This is happening to me, too. 
I have been trying to get metrics data of about 130 people but it simply teturns empty array.
Is it some sort of settings in IDE?
We can get the metrics from organization dimention, which has exactly the same member as the team, though.

**Comment by NamBarthur** [comment]
I see discrepancies between the metrics for enterprise level and metrics for team level. I see there are gaps in data captured for team level metrics which are not observed in enterprise level. I wanted to understand if there are any future fixes being made available for the team API.

**Comment by rm-benniejohnston** [comment]
Any update on when it will publish metrics for Edit mode as well ?
